### DISCUSSION TITLE
Displaying text in sprite as HEX

### DISCUSSION THREAD

**Discussion by beng2k** [Bodmer/TFT_eSPI#2329] [Q&A - Sprites]
Hi,

I am currently receiving data in decimal form and being written to an array.
I can display in serial by using print(dataValues[1], HEX)

How can I display the text in the sprite in HEX format?
Also noticed when the data changes from 4 digits to 3 digit, the 4th digit remains.  Is there something I am missing for only showing the current value?

`void updateARBID()
{
    frameid.setTextColor(TFT_WHITE, TFT_BLUE, true);
    frameid.drawNumber(dataValues[1], 40,20,2);
    frameid.pushSprite(20, 40);
    delay(200);
}`

**Comment by Bodmer** [comment]
You can use spr.setCursor(x, y); and spr.print(value,HEX);

**Reply by beng2k** [reply]
Thanks for the quick response.

I tried as above however when i set the postion to 1,1 it continues to write new values beside the old values until the sprite is full and nothing more is added, at least visible.  

    if (arb2msgData[2] != oldarb2msgData[2])
      {
        arb2code1.setTextColor(TFT_WHITE, TFT_RED, true);
      }
      else
      {
        arb2code1.setTextColor(TFT_WHITE, TFT_BLACK, true);
      }  
    arb2code0.setCursor(1,1);
    arb2code1.print(arb2msgData[2], HEX);
    arb2code1.pushSprite(120, 80);

This is the code i am using.  Any suggestions?
